reactor-netty-core/src/main/java/reactor/netty/channel/FluxReceive.java

```diff
@@ -182,11 +183,19 @@ final void startReceiver(CoreSubscriber<? super Object> s) {
 			}
 			else {
 				if (log.isDebugEnabled()) {
-					log.debug(format(channel, "{}: Only one connection receive subscriber allowed."), this);
+					log.debug(format(channel, "{}: Ignoring additional inbound receiver subscriber."), this);
 				}
-				Operators.error(s,
-						new IllegalStateException(
-								"Only one connection receive subscriber allowed."));
+
+				StringBuilder msg = new StringBuilder("Ignoring additional inbound receiver subscriber.");
+				msg.append(" Current flux state=[");
+				msg.append("terminated=").append(inboundDone);
+				msg.append(",cancelled=").append(isCancelled());
+				msg.append(",pending=").append(pending);
+				msg.append(",error=").append(inboundError != null);
+				msg.append("].");
+				IllegalStateException ex = inboundError == null ? new IllegalStateException(msg.toString()) :
+						new IllegalStateException(msg.toString(), inboundError);
```

Comment:
Maybe the same (and more complete) message should also be used for the log message above?

Reply:
the same kind of informations is actually logged in the log message above, using the FluxReceive.toString() method (which displays the same informations a bit differently)